jme3-effects/src/main/java/com/jme3/post/filters/PBRBloomFilter.java

```diff
@@ -0,0 +1,262 @@
+            } else {
+                prev = upsamplingPasses[i-1];
+            }
+            Pass pass = new Pass() {
+                @Override
+                public void beforeRender() {
+                    upsampleMat.setTexture("Texture", prev.getRenderedTexture());
+                    upsampleMat.setVector2("TexelSize", texelSize);
+                }
+            };
+            pass.init(renderer, w, h, format, Image.Format.Depth, 1, upsampleMat);
+            postRenderPasses.add(pass);
+            upsamplingPasses[i] = pass;
+        }
+        
+        material = new Material(assetManager, "Common/MatDefs/Post/PBRBloomFinal.j3md");
+        material.setTexture("GlowMap", upsamplingPasses[upsamplingPasses.length-1].getRenderedTexture());
+        
+        initialized = true;
+        
+    }
+    
+    @Override
+    protected Material getMaterial() {
+        return material;
+    }
+    
+    /**
+     * Sets the number of sampling passes in each step.
+     * <p>
+     * Higher values produce more glow with higher resolution, at the cost
+     * of more passes. Lower values produce less glow with lower resolution.
+     * <p>
+     * The total number of passes is {@code 2n+1}: n passes for downsampling
+     * (13 texture reads per pass per fragment), n passes for upsampling and blur
+     * (9 texture reads per pass per fragment), and 1 pass for blending (2 texture reads
+     * per fragment). Though, it should be noted that for each downsampling pass the
+     * number of fragments decreases by 75%, and for each upsampling pass, the number
+     * of fragments quadruples (which restores the number of fragments to the original
+     * resolution).
+     * <p>
+     * Settings this after the filter has been initialized forces reinitialization.
+     * <p>
+     * default=5
+     * 
+     * @param n number of passes per donwsampling/upsampling step
+     */
+    public void setNumSamplingPasses(int n) {
+        if (n <= 0) {
+            throw new IllegalArgumentException("Expected number of sampling passes to be greater than zero (found: "+n+").");
+        }
+        numSamplingPasses = n;
+        if (initialized) {
+            initFilter(assetManager, renderManager, viewPort, width, height);
+        }
+    }
+    
+    /**
+     * Sets the factor at which the glow result texture is merged with
+     * the scene texture.
+     * <p>
+     * Low values favor the scene texture more, while high values make
+     * glow more noticable. This value is clamped between 0 and 1.
+     * <p>
+     * default=0.05f
+     * 
+     * @param factor 
+     */
+    public void setGlowFactor(float factor) {
+        this.glowFactor = FastMath.clamp(factor, 0, 1);
+        if (material != null) {
+            material.setFloat("GlowFactor", glowFactor);
+        }
+    }
+    
+    /**
+     * Gets the number of downsampling/upsampling passes per step.
+     * 
+     * @return glow factor
```

Comment:
"`@return glow factor`": error in the javadoc
should be "`@return number of downsampling/upsampling passes`"